src/ansi.rs

```diff
@@ -363,46 +343,29 @@ impl<T: Timeout> Processor<T> {
     {
         self.state.sync_state.buffer.push(byte);
 
-        // Handle sync DCS escape sequences.
-        match self.state.sync_state.pending_dcs {
-            Some(_) => self.advance_sync_dcs_end(handler, byte),
-            None => self.advance_sync_dcs_start(),
-        }
+        // Handle sync CSI escape sequences.
+        self.advance_sync_csi(handler);
     }
 
-    /// Find the start of sync DCS sequences.
-    fn advance_sync_dcs_start(&mut self) {
+    /// Handle BSU/ESU CSI sequences during synchronized update.
+    fn advance_sync_csi<H>(&mut self, handler: &mut H)
+    where
+        H: Handler,
+    {
         // Get the last few bytes for comparison.
         let len = self.state.sync_state.buffer.len();
-        let offset = len.saturating_sub(SYNC_ESCAPE_START_LEN);
+        let offset = len.saturating_sub(SYNC_ESCAPE_LEN);
         let end = &self.state.sync_state.buffer[offset..];
 
+        // NOTE: It is technically legal to specify multiple private modes in the same
+        // escape, but we only allow EXACTLY `\e[?2026h`/`\e[?2026l` to keep the parser
+        // reasonable.
```

Comment:
I've tested this with Kitty (and read their impl) and they do the same. Seems like somewhat of an oversight of the protocol specification.